_posts/2020-11-30-developer-survey-2020.md

```diff
@@ -0,0 +1,99 @@
+
+When asked what platforms we would like to use in the future, the top 3 consisted of the three big cloud services. The most wanted was AWS, followed by Azure and then Google Cloud Platform. This mirrored the current usage.
+
+## IDEs
+
+![Editors](/img/developer-survey-2020/editors.png)
+
+A new question this year asked what IDEs or editors we use. Most people use several, perhaps based on what languages they are working with currently. Of course then there are also those who only use one (Emacs people wouldn't probably want to leave it for any task they need to do). All the cloud services had their usage increase compared to 2018 survey.
+
+Most popular one was IntelliJ Idea and other IDEs based on it. Idea is definitely very liked for Java work and Cursive for Clojure and Android Studio for Android app development are much used too. Visual Studio Code has gathered a huge following in a relatively short time and it's no surprise being used a lot among our developers too, considering that [StackOverflow survey in 2019 ](https://insights.stackoverflow.com/survey/2019#development-environments-and-tools) had it as the most popular editor.
+
+The eternal war between VIM and Emacs users ended this time in favor of Vim, as Vim was used by 9% and Emacs was used by 7%.
+
+## Operating Systems
+
+![Operating systems](/img/developer-survey-2020/operating_systems.png)
+
+Over half of the respondents reported using macOS as their primary operating system. About a third reported using Windows and a bit over 10% used Linux. A few people reported using multiple operating systems and mostly working in virtual machines. Virtual machine usage for different use cases is probably more common than that, as the question was about the primary operating system in use and not about every operating system people use in some capacity.
+
+## Trends
+
+Usage of cloud platforms is on the rise and that has an impact on lots of things, like solutions used for infrastructure-as-code or CI/CD. For example, Jenkins was still the most used CI/CD solution, but its usage had dropped a quite significantly in percentage and the usage of the solutions provided by the cloud platforms, such as Code* tools in AWS were up. In infrastructure-as-code tools Ansible was still the most used, but its usage had dropped a lot and AWS CDK, Terraform etc. had a lot more usage now than two years ago (TARKISTA!!!).
+
+TypeScript usage has increased significantly in two years. When asking for generally used programming languages, TypeScript was reported by 35% of the respondents, where in 2018 it was only used by 19%. This change also meant that JavaScript,even though still the most widely used language in general, dropped from 84% to 69% of usage.
+
+## Conclusion
```

Comment:
Could be interesting to see how our results compare to other surveys such as StackOverflow's. But maybe that is too much work 😄

Reply:
Good suggestion. I'll take a look if it's possible for some questions, but overall I think most questions are too different between the surveys. Especially with the newest Stackoverflow survey I think there were quite a big changes compared to the old one (and answer choices were in some parts quite controversial).